php/Exchange.php

```diff
@@ -141,15 +141,15 @@ public static function decimal ($number) {
     }
 
     public static function safe_float ($object, $key, $default_value = null) {
-        return (is_array ($object) && array_key_exists ($key, $object) && $object[$key]) ? floatval ($object[$key]) : $default_value;
+        return (isset ($object[$key]) && is_numeric ($object[$key])) ? floatval ($object[$key]) : $default_value;
```

Comment:
A quick question: the safe_float function should coerce to float, wouldn't `is_numeric()` prevent that in this particular place?

Reply:
If it's not a valid float number, then there's no point to bother with it, and fallback to `$default_value`, which is not a float either by default.

Reply:
Or if you mean that `is_numeric` is only for integers, then it can handle floats too.

Reply:
No, nvm, I forgot it works ok with strings.